Recast/Source/RecastArea.cpp

```diff
@@ -75,8 +58,8 @@ bool rcErodeWalkableArea(rcContext* ctx, int radius, rcCompactHeightfield& chf)
 						{
 							const int nx = x + rcGetDirOffsetX(dir);
 							const int ny = y + rcGetDirOffsetY(dir);
-							const int nidx = (int)chf.cells[nx+ny*w].index + rcGetCon(s, dir);
-							if (chf.areas[nidx] != RC_NULL_AREA)
+							const int nidx = (int)chf.cells[nx + ny*w].index + rcGetCon(s, dir);
+							if (chf.areas[nidx] != fromAreaID)
```

Comment:
Well, no, this block is only entered if `rcGetCon` does _not_ return `RC_NOT_CONNECTED` (see line 57 in your commit), so `nidx` is safe to use. However, the variable `nc` can only be incremented in this block; in other words, the only way it can equal 4 on line 69 is if all four neighbors are connected.